src/wallet/test/wallet_transaction_tests.cpp

```diff
@@ -4,12 +4,187 @@
 
 #include <wallet/transaction.h>
 
+#include <wallet/coincontrol.h>
+#include <kernel/chain.h>
+#include <validation.h>
+#include <wallet/receive.h>
+#include <wallet/spend.h>
+#include <wallet/test/util.h>
 #include <wallet/test/wallet_test_fixture.h>
 
 #include <boost/test/unit_test.hpp>
 
 namespace wallet {
-BOOST_FIXTURE_TEST_SUITE(wallet_transaction_tests, WalletTestingSetup)
+BOOST_FIXTURE_TEST_SUITE(wallet_transaction_tests, BasicTestingSetup)
+
+const CWalletTx* GenBlockAndRetrieveMinedTx(TestChain100Setup& context, const CMutableTransaction& tx_to_mine, const CScript& coinbase_script, CWallet& wallet)
+{
+    const CBlock& block = context.CreateAndProcessBlock({tx_to_mine}, coinbase_script);
+    wallet.blockConnected(kernel::MakeBlockInfo(WITH_LOCK(cs_main, return context.m_node.chainman->ActiveChain().Tip()), &block));
+    return Assert(WITH_LOCK(wallet.cs_wallet, return wallet.GetWalletTx(tx_to_mine.GetHash())));
+}
+
+const CWalletTx* SendCoinsAndGenBlock(TestChain100Setup& context, const CScript& scriptpubkey, CAmount amount, CWallet& from_wallet, const CScript& coinbase_script)
+{
+    CCoinControl coin_control;
+    auto op_tx = Assert(CreateTransaction(from_wallet, {{scriptpubkey, amount, true}}, 1, coin_control));
+    return GenBlockAndRetrieveMinedTx(context, CMutableTransaction(*op_tx->tx), coinbase_script, from_wallet);
+}
+
+void CreateMultisigScriptAndImportDescriptor(CWallet& wallet, CScript& multisig_script_out, std::vector<CKey>& priv_keys)
+{
+    auto spkm = static_cast<DescriptorScriptPubKeyMan*>(wallet.GetScriptPubKeyMan(OutputType::BECH32, false));
+    std::vector<CPubKey> pks;
+    std::string pks_str; // descriptor data
+    for (int i=0; i < 5; i++) {
+        CTxDestination multi_dest = *Assert(spkm->GetNewDestination(OutputType::BECH32));
+        auto provider = Assert(spkm->GetSolvingProvider(GetScriptForDestination(multi_dest), /*include_private=*/ true));
```

Comment:
I would strongly prefer that we don't have to add new methods to DescSPKM that can export individual keys. This test could just as easily work with generating new random CKeys for the private keys.